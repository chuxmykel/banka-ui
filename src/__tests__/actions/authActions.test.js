import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  signUp,
  logIn,
  logOut,
} from '@Actions/authActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-bank.herokuapp.com/api/v1';

describe('Auth action tests', () => {
  describe('Auth Action tests return values', () => {
    const userData = {
      firstName: 'john',
      lastName: 'doe',
      email: 'john.doe@gmail.com',
      password: 'password',
    };

    const history = {
      push: jest.fn(),
    };

    it('signup should return expected values', () => {
      expect(signUp(userData, history)).toMatchSnapshot();
    });

    it('signin should return expected values', () => {
      expect(logIn(userData, history)).toMatchSnapshot();
    });

    it('logout should return expected values', () => {
      expect(logIn(history)).toMatchSnapshot();
    });
  });

  describe('Async action creators test', () => {
    let store;
    const response = {
      data: [{
        id: 1,
        token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NSwiZW1haWwiOiJrY215a2FpcmxAZ21haWwuY29tIiwiZmlyc3ROYW1lIjoiS2VsZWNoaSIsImxhc3ROYW1lIjoiTmd3b2JpYSIsInR5cGUiOiJjbGllbnQiLCJpc0FkbWluIjpudWxsLCJpYXQiOjE1NjY3NjgzMDIsImV4cCI6MTU2Njg1NDcwMn0.tbi_p7QnWb524thZ6uao7ILrxt0Vya_JCec1skuoGjE',
        firstName: 'John',
      }],
    };

    beforeEach(() => {
      store = mockStore({});
    });

    afterEach(() => {
      nock.cleanAll();
    });

    it('signs up a user successfully', () => {
      nock(url)
        .post('/auth/signup')
        .reply(201, response);

      return store.dispatch(signUp({}, { push: jest.fn() }))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('signs in a user successfully', () => {
      nock(url)
        .post('/auth/signin')
        .reply(200, response);

      return store.dispatch(logIn({}, { push: jest.fn() }))
        .then(() => {
          expect(store.getActions()).toMatchSnapshot();
        });
    });

    it('logs out a user successfully', () => {
      store.dispatch(logOut({ push: jest.fn() }));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
