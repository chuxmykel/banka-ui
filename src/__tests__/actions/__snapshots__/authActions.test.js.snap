// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth action tests Async action creators test logs out a user successfully 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "SET_CURRENT_USER",
  },
]
`;

exports[`Auth action tests Async action creators test signs in a user successfully 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
  Object {
    "payload": Object {
      "email": "kcmykairl@gmail.com",
      "exp": 1566854702,
      "firstName": "Kelechi",
      "iat": 1566768302,
      "id": 5,
      "isAdmin": null,
      "lastName": "Ngwobia",
      "type": "client",
    },
    "type": "SET_CURRENT_USER",
  },
  Object {
    "payload": "signin",
    "type": "MODAL_CLOSE",
  },
]
`;

exports[`Auth action tests Async action creators test signs up a user successfully 1`] = `
Array [
  Object {
    "type": "AUTHENTICATING",
  },
  Object {
    "payload": Object {
      "email": "kcmykairl@gmail.com",
      "exp": 1566854702,
      "firstName": "Kelechi",
      "iat": 1566768302,
      "id": 5,
      "isAdmin": null,
      "lastName": "Ngwobia",
      "type": "client",
    },
    "type": "SET_CURRENT_USER",
  },
  Object {
    "payload": "signup",
    "type": "MODAL_CLOSE",
  },
]
`;

exports[`Auth action tests Auth Action tests return values logout should return expected values 1`] = `[Function]`;

exports[`Auth action tests Auth Action tests return values signin should return expected values 1`] = `[Function]`;

exports[`Auth action tests Auth Action tests return values signup should return expected values 1`] = `[Function]`;
